pkg/controller/installation/products/rhssouser/reconciler.go

```diff
@@ -0,0 +1,365 @@
+		},
+	}
+}
+
+// Reconcile reads that state of the cluster for rhsso and makes changes based on the state read
+// and what is required
+func (r *Reconciler) Reconcile(ctx context.Context, inst *v1alpha1.Installation, product *v1alpha1.InstallationProductStatus, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	ns := r.Config.GetNamespace()
+
+	phase, err := r.ReconcileNamespace(ctx, ns, inst, serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.ReconcileSubscription(ctx, inst, marketplace.Target{Pkg: defaultSubscriptionName, Channel: marketplace.IntegreatlyChannel, Namespace: r.Config.GetNamespace()}, serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.reconcileComponents(ctx, inst, serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.handleProgressPhase(ctx, serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	product.Host = r.Config.GetHost()
+	product.Version = r.Config.GetProductVersion()
+
+	r.logger.Infof("%s has reconciled successfully", r.Config.GetProductName())
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) reconcileComponents(ctx context.Context, inst *v1alpha1.Installation, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	r.logger.Info("Reconciling Keycloak components")
+	kc := &aerogearv1.Keycloak{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      keycloakName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	ownerutil.EnsureOwner(kc, inst)
+	or, err := controllerutil.CreateOrUpdate(ctx, serverClient, kc, func(existing runtime.Object) error {
+		kc := existing.(*aerogearv1.Keycloak)
+		kc.Spec.Plugins = []string{
+			"keycloak-metrics-spi",
+			"openshift4-idp",
+		}
+		kc.Spec.Provision = true
+		return nil
+	})
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to create/update keycloak custom resource")
+	}
+	r.logger.Infof("The operation result for keycloak %s was %s", kc.Name, or)
+
+	kcr := &aerogearv1.KeycloakRealm{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      keycloakRealmName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	ownerutil.EnsureOwner(kcr, inst)
+	or, err = controllerutil.CreateOrUpdate(ctx, serverClient, kcr, func(existing runtime.Object) error {
+		kcr := existing.(*aerogearv1.KeycloakRealm)
+		kcr.Spec.CreateOnly = false
+		kcr.Spec.BrowserRedirectorIdentityProvider = idpAlias
+
+		if kcr.Spec.KeycloakApiRealm == nil {
+			kcr.Spec.KeycloakApiRealm = &aerogearv1.KeycloakApiRealm{}
+		}
+		kcr.Spec.ID = keycloakRealmName
+		kcr.Spec.Realm = keycloakRealmName
+		kcr.Spec.DisplayName = keycloakRealmName
+		kcr.Spec.Enabled = true
+		kcr.Spec.EventsListeners = []string{
+			"metrics-listener",
+		}
+
+		if kcr.Spec.Users == nil {
+			kcr.Spec.Users = []*aerogearv1.KeycloakUser{CustomerAdminUser}
```

Comment:
There is a ticket to remove customer-admin user so we can remove that from here:
https://issues.jboss.org/browse/INTLY-2932